src/vs/code/browser/workbench/workbench.ts

```diff
@@ -508,4 +508,11 @@ function doCreateUri(path: string, queryValues: Map<string, string>): URI {
 		urlCallbackProvider: new LocalStorageURLCallbackProvider(config.callbackRoute),
 		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider() // with a remote, we don't use a local credentials provider
 	});
+
+	if ('serviceWorker' in navigator) {
```

Comment:
Would there be a way to do this only when running as a PWA to avoid a service worker if running in a browser tab?

Reply:
Yes.
When a page is running as a PWA it's CSS display-mode is standalone, so `window.matchMedia('(display-mode: standalone)').matches || navigator.standalone`.
But a page has to register a service worker *before* it is installed, otherwise Chrome doesn't display the Install button at all.

Reply:
Yes.
``` javascript
window.addEventListener('DOMContentLoaded', () => {
	if (window.matchMedia('(display-mode: standalone)').matches || navigator.standalone) {
		// PWA
		window.matchMedia('(display-mode: standalone)').addListener((ev) => {
			if (ev.matches) {
				// PWA installed
			}
		});
	} else {
		// webpage
	}
});
```
But a page must register a service worker *before* it is installed, otherwise Chrome doesn't display the Install button at all.